### DISCUSSION TITLE
Pixel perfect detection with BaseRenderTexture

### DISCUSSION THREAD

**Discussion by KennethGomez** [pixijs/pixijs#7356] [Q&A] [answered]
I've seen an example of pixel perfect detection but it does not work at all with BaseRenderTexture since its resource property is always null.

How can I solve this?

**Comment by ivanpopelyshev** [comment, marked as answer]
Use `extract` in pixi to get data  when you update the renderTexture, i hope its not often :) `extract` uses webgl `readPixels` operation.

However, `readPixels` is a slow operation that should be done in the end of frame, that's why renderTexture's are not suited for hit detection.
